lib/packages/touch.rb

```diff
@@ -1,22 +1,25 @@
 # simple implementation of the 'touch' command, which can be used to create empty text files
 # Initially, this can be used for simple file creation in shellrb. Eventually, I would like to allow for access time manipulation
 require_relative 'ls'
+require 'set'
 
 class Touch
   def self.main params
     # user passes the name of the file to be created as the first parameter
     new_file = params[0]
     
-    # gets a list of the files and directories currently in the directory
-    
-    ls = Ls.show_hidden
+    # creates a set of all the files in the current directory
+    ls = Set.new
+    ls.add Ls.show_hidden
+    ls.add Ls.show_non_hidden
+
     file_present = false
     
     # iterates through the files in the current directory to make sure that the desired file does not exist
     ls.each do |x|
-      if new_file == x
-          file_present = true
-      end
+      if x.include? new_file
+        file_present = true
```

Comment:
add a `break` command in case the file is found, else it will keep checking for the file even if the file is found